common/JackAtomic.h

```diff
@@ -21,6 +21,15 @@ Foundation, Inc., 59 Temple Place - Suite 330, Boston, MA 02111-1307, USA.
 #define __JackAtomic__
 
 #include "JackTypes.h"
+
+#ifdef WIN32
+#include "atomic_msvc.h"
+#define COMPARE_EXCHANGE(ADDRESS, NEW, EXPECTED) atomic::msvc::interlocked<UInt32, 4>::compare_exchange(ADDRESS, NEW, EXPECTED)
+#else
+#define COMPARE_EXCHANGE(ADDRESS, NEW, EXPECTED) CAS(EXPECTED, NEW, ADDRESS)
+#endif
+
+#ifndef WIN32
```

Comment:
Firstly, WIN32 should probably be replaced by _WIN32 everywhere. Secondly, this one should be "#ifndef !_MSC_VER", not "#ifndef WIN32".

Reply:
I started off with wrong pre-processor flags, and changed my mistakes later. I'll fix this one! And I may have been too aggrissve towards MSVC, should keep mingw in mind where it makes sense. Maybe inline assembly could be a bad idea anyway? But I'll keep my scope on MSVC.